eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -55,12 +55,13 @@ $noDeleteAfter | ForEach-Object { Write-Verbose $_.ResourceGroupName }
 
 $hasDeleteAfter = $allGroups.Where({ $_.Tags.Keys -contains "DeleteAfter" })
 Write-Host "Count $($hasDeleteAfter.Count)"
-$toDelete = $hasDeleteAfter.Where({ $now -gt [DateTime]$_.Tags.DeleteAfter })
+$toDelete = $hasDeleteAfter.Where({ $deleteDate = ($_.Tags.DeleteAfter -as [DateTime]); (!$deleteDate -or $now -gt $deleteDate) })
```

Comment:
Why not just use:

```powershell
[DateTime] $deleteAfter = $null
$toDelete = $hasDeleteAfter.Where({ [DateTime]::TryParse($_.Tags.DeleteAfter, [ref] $deleteAfter) -and $deleteAfter -lt [DateTime]::Now })
```

Reply:
I want to do it for each and I personally like `-as` operator for doing the parsing it will either parse or be null and no need for verbose ref parameters.